pkg/kubelet/kuberuntime/kuberuntime_container_windows.go

```diff
@@ -76,12 +104,6 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
 		wc.Resources.CpuMaximum = cpuMaximum
```

Comment:
The actual situation `NanoCpus` should have nothing to do with the number of` cpu`, its correct calculation method should be like this `NanoCpus  = cpuLimit.MilliValue() * (1e9 / 10000)`

I have a question is, does it seem that `CpuMaximum` means the same as` NanoCpus`?

This way can really avoid modifying the cri 👍

Reply:
Good catch! I should have read my own comments again.

The original container limit isn't available in Dockershim, just the CRI representation (CpuMaximum). This will be a good discussion point with @jterry75 and @Random-Liu on if CRI-API should be using units that are relative to the node's core count (CpuMaximum), or absolute (NanoCPUs).

Reply:
Not sure I understand. CRI very clearly defines this as "count". So its full cpu's relative to the nodes core count. What is CRI doing wrong?

Reply:
@jterry75 cpucount cannot precisely control cpulimit, it can only be limited to an integer number of cpu

Reply:
I also have a question, if the `NanoCPUs` passed to docker after this calculation is a corrected value instead of the original value.

Reply:
CpuCount = full cores
CpuMaximum = 1/10000th of the total cores in the node.
NanoCPUs = 1e-9 of a single core

Reply:
This implementation is simpler and more consistent than modifying the CRI 👍

Reply:
@PatrickLang - I think this case is wrong. The hypervisor "Maximum" is based on the vCPU scheduler. So for a request of say .5 -> 1 vCPU Limit 50% but a request of 1.25 -> 2 vCPU Limit 62.5%.

Reply:
Alright - I'm keeping this as-is and added tests.

Reply:
trying to make sure i understand it, so CpuShares is not used anymore (removed), CpuCount is used for HyperV (not for process isolation), CpuMaximum is used for both HyperV and process isolation.

one question: since these three are mutually exclusive, then for hyperV case, what is the point of setting both CpuCount and CpuMaximum?

Reply:
Hyper-V still needs a CPU count because it could for example give 2 cores, but throttle each to 50%.

Hyper-V needs a redesign - this code is not tested and I don't think it works. I'll probably delete it in a separate PR as I'm trying to keep this one narrowly-scoped so it can be backported.